opsmngr/agent_api_keys.go

```diff
@@ -57,16 +55,17 @@ func (s *AgentAPIKeysServiceOp) Create(ctx context.Context, projectID string) (*
 }
 
 // See more: https://docs.opsmanager.mongodb.com/current/reference/api/agentapikeys/get-all-agent-api-keys-for-project/
-func (s *AgentAPIKeysServiceOp) List(ctx context.Context, projectID string) (*AgentAPIKeys, *atlas.Response, error) {
+func (s *AgentAPIKeysServiceOp) List(ctx context.Context, projectID string) (*[]AgentAPIKey, *atlas.Response, error) {
 	path := fmt.Sprintf(agentAPIKeysBasePath, projectID)
 
 	req, err := s.Client.NewRequest(ctx, http.MethodGet, path, nil)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	root := new(AgentAPIKeys)
+	root := new([]AgentAPIKey)
```

Comment:
this is a bit of a gotcha with go, for arrays you should use `make` instead of new, this goes along with my previous comment of making the type `[]*` and not `*[]` but I do expect the line `s.Client.Do(ctx, req, root)` to be picky about  this, probably may need to be `s.Client.Do(ctx, req, &root)` so let's make sure this works